### PR TITLE
[GH-26382] format command mmctl auth list

### PR DESCRIPTION
**Pull request by annaos**
#### Summary

- Fixing the formatting of `mmctl auth list` if the username which was used to connect mmctl to Mattermost is too short
- Adding a test for the changes.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/26382
Jira https://mattermost.atlassian.net/browse/MM-57075

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/cmd/mmctl/commands/auth_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+package commands
+
+import (
+	"github.com/mattermost/mattermost/server/v8/cmd/mmctl/printer"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"os"
+	"os/user"
+	"path/filepath"
+)
```
Comment (hanzei) [inline server/cmd/mmctl/commands/auth_test.go:13]: CI is unhappy with the import order: 
```
/home/runner/work/mattermost/mattermost/server/bin/golangci-lint run ./...
cmd/mmctl/commands/auth_test.go:7: File is not `goimports`-ed (goimports)
	"github.com/mattermost/mattermost/server/v8/cmd/mmctl/printer"
	"github.com/spf13/cobra"
	"github.com/spf13/viper"
make: *** [Makefile:289: golangci-lint] Error 1
```

Please use the same order as in https://github.com/annaos/mattermost/blob/ee6be981c952791fe2de556e9964d47e2b77e796/server/cmd/mmctl/commands/channel_users_e2e_test.go#L6-L16.

File: server/cmd/mmctl/commands/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func (s *MmctlUnitTestSuite) TestAuthList() {
+	originalUser := *currentUser
+	defer func() {
+		SetUser(&originalUser)
+	}()
+
+	tmp, _ := os.MkdirTemp("", "mmctl-")
```
Comment (hanzei) [inline server/cmd/mmctl/commands/auth_test.go:21]: Let's check the error

File: server/cmd/mmctl/commands/auth_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	defer os.RemoveAll(tmp)
```
Comment (hanzei) [inline server/cmd/mmctl/commands/auth_test.go:22]: Nit pick: In general, it's advised to use `s.T().Cleanup` to cleanup after the test run.

**Review by hanzei — APPROVED** [review]
These tests are awesome! Thanks your :rocket:

**Review by hanzei — APPROVED** [review]

**Review by vish9812 — APPROVED** [review]
Nice work :+1: 
Thanks @annaos

**Comment by hanzei** [comment]
Tested locally and confirmed that even with short usernames or siteURLs the table is still correctly formated. @DHaussermann Are we good to merge the PR?

**Review by DHaussermann — APPROVED** [review]
Tested and passed 
- Tested the case where the list table shows very short username
- This has been resolved and the value is aligned in the cell and no table structure is broken
- Regression tested cases where the tables has multiple rows of data
- Tested various lengths for username and ensured connection name works as well if it's only 1 or 2 characters long 
LGTM!

Huge thanks @annaos for this fix! 🎉

**Pull request merged**